File: 프로그래머스/unrated/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    score = {1:3,2:2,3:1,4:0,5:1,6:2,7:3}
    mbti = {"R":0,"T":0, "C":0,"F":0, "J":0,"M":0, "A":0,"N":0}
    for m in range(len(survey)):
        data1 = survey[m][0]
        data2 = survey[m][1]
        if choices[m] > 4:
            mbti[data2] += score[choices[m]]
        elif choices[m] < 4:
            mbti[data1] += score[choices[m]]
    # print(mbti)
    answer += "R" if mbti["R"] >= mbti["T"] else "T"
    answer += "C" if mbti["C"] >= mbti["F"] else "F"
    answer += "J" if mbti["J"] >= mbti["M"] else "M"
    answer += "A" if mbti["A"] >= mbti["N"] else "N"
    
    return answer

File: 프로그래머스/unrated/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        HashMap<String,Integer> map = new HashMap<>();
        map.put("R",0);
        map.put("T",0);
        map.put("C",0);
        map.put("F",0);
        map.put("J",0);
        map.put("M",0);
        map.put("A",0);
        map.put("N",0);
        for(int i=0; i<survey.length; i++){
            String[] m = survey[i].split("");
            if(choices[i] > 4){
                map.put(m[1],map.get(m[1])+(choices[i] - 4));
            }
            else if(choices[i] < 4){
                map.put(m[0],map.get(m[0])+(4-choices[i]));
            }
        }
        if(map.get("R") >= map.get("T")){
            answer +="R";
        }
        else{
            answer+="T";
        }
        if(map.get("C") >= map.get("F")){
            answer +="C";
        }
        else{
            answer+="F";
        }
        if(map.get("J") >= map.get("M")){
            answer +="J";
        }
        else{
            answer+="M";
        }
        if(map.get("A") >= map.get("N")){
            answer +="A";
        }
        else{
            answer+="N";
        }
        
        return answer;
    }
}